app.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+# !/usr/bin/env ruby
+require 'sinatra'
+require 'sinatra/reloader'
+require 'erb'
+require 'json'
+
+helpers do
+  include ERB::Util
+
+  def sanitize(text)
+    escape_html(text)
+  end
+
+  def read_memos
+    memos = Dir.glob('*.json', base: 'data').map do |file_name|
+      read_memo(file_name.delete('.json'))
+    end
+    memos.sort_by { |memo| memo[:time] }.reverse
+  end
+
+  def read_memo(id)
+    JSON.parse(File.open("data/#{id}.json").read, symbolize_names: true)
+  end
+
+  def save_memo(id, memo)
+    File.open("data/#{id}.json", 'w') do |file|
+      JSON.dump(memo, file)
+    end
+  end
+
+  def delete_memo(id)
+    File.delete("data/#{id}.json")
+  end
+end
+
+get '/memos' do
+  @memos = read_memos
+  erb :top
+end
+
+get '/new' do
+  erb :new
+end
+
+post '/memos' do
+  memo = { id: SecureRandom.uuid, time: Time.now, title: params[:title], content: params[:content] }
+  save_memo(memo[:id], memo)
+  redirect to('/memos')
+end
+
+get '/memos/:id' do |id|
+  if File.exist?("data/#{id}.json")
+    @memo = read_memo(id)
+    erb :details
+  else
+    erb :not_found
+  end
```

Comment:
追加してもらったファイルの存在確認ですが、改めて考えると今のロジックであればセキュリティ上の脆弱性も発生しなさそうなので、無くしちゃった方がシンプルになるなと思いました。

```suggestion
  @memo = read_memo(id)
  erb :details
```

他の処理も同様に対処して、`not_found.erb`は削除で良いと思います 🙏